### ISSUE TITLE
Is it possible to let conv net running on tensorflow to control autonomous driving RC car on Raspberry Pi?

### ISSUE THREAD

**Issue by zxzhijia** [samjabrahams/tensorflow-on-raspberry-pi#17]
Is it possible to let conv net running on tensorflow to control autonomous driving RC car on Raspberry Pi?

**Comment by zxzhijia** [comment]
The car receiving camera images and output steering angles.

**Comment by samjabrahams** [comment]
I don't see why it couldn't work. You could use the `picamera` module and techniques similar to those found in [this OpenCV example](http://www.pyimagesearch.com/2015/03/30/accessing-the-raspberry-pi-camera-with-opencv-and-python/).  Then, use the [RPIO](https://pythonhosted.org/RPIO/) module to convert model output into GPIO instructions (which would hook up to your RC car controller or something similar).

The main challenges would be collecting enough accurate training data, as well as creating a model compact enough to evaluate quickly on the Raspberry Pi. For the training data, you'd ideally have a way to capture film and the steering angles (as controlled by a human) in real time. To create a fast model, you'll have to play around with different architectures to see what is the best trade-off between accuracy and evaluation time.

There are a few of us working on making these TensorFlow binaries more efficient, as well as try out the recently introduced [8-bit quantization capabilities](https://petewarden.com/2016/05/03/how-to-quantize-neural-networks-with-tensorflow/) to try and speed up evaluation, so hopefully these sorts of RPi models will be able to get more sophisticated over time.

**Comment by zxzhijia** [comment]
That's great. Thank you very much.

**Comment by samjabrahams** [comment]
No problem! It'd be great to get an update on how it turns out down the road.

Closing this, as I believe the question is answered.

**Comment by yufengg** [comment]
it appears the author got it working(!) -- https://github.com/zxzhijia/GoPiGo-Driven-by-Tensorflow
